recovery/elliptics_recovery/types/dc.py

```diff
@@ -234,18 +250,84 @@ def contains(addresses_with_backends, address, backend_id):
     return address_range
 
 
-def final_merge(ctx, results):
+def process_uncommitted(ctx, results):
+    if ctx.dry_run or ctx.safe:
+        return
+
+    elog = elliptics.Logger(ctx.log_file, int(ctx.log_level))
+    node = elliptics_create_node(address=ctx.address,
+                                 elog=elog,
+                                 wait_timeout=ctx.wait_timeout,
+                                 flags=elliptics.config_flags.no_route_list,
+                                 net_thread_num=1,
+                                 io_thread_num=1,
+                                 remotes=ctx.remotes)
+    session = elliptics.newapi.Session(node)
+    session.trace_id = ctx.trace_id
+    session.exceptions_policy = elliptics.exceptions_policy.no_exceptions
+    session.set_filter(elliptics.filters.all_final)
+    session.ioflags |= elliptics.io_flags.cas_timestamp
+    session.timestamp = ctx.prepare_timeout
+
+    for filename, uncommitted_filename, _, _ in results:
+        with open(filename, 'ab') as f:
+            for _, batch in groupby(enumerate(load_key_data(uncommitted_filename)),
+                                    key=lambda x: x[0] / ctx.batch_size):
+                batch = [item[1] for item in batch]
+                tasks = []
+                for key, key_infos in batch:
+                    for info in key_infos:
+                        if info.flags & elliptics.record_flags.uncommitted:
+                            tasks.append((key, info.group_id))
+
+                statuses = {} # (key, group_id) -> status
+                for attempt in range(ctx.attempts):
+                    if not tasks:
+                        break
+
+                    responses = []
+                    for key, group_id in tasks:
+                        session.groups = [group_id]
+                        responses.append(session.remove(key))
+
+                    failed_tasks = []
+                    is_last_attempt = attempt == ctx.attempts - 1
+                    for i, r in enumerate(responses):
+                        key, group_id = tasks[i]
+                        status = r.get()[0].status
+                        log.info('Removed uncommitted key: %s, group: %s, status: %s, last attempt: %s',
+                                 key, group_id, status, is_last_attempt)
+                        statuses[(key, group_id)] = status
+                        if status in (-errno.ETIMEDOUT, -errno.ENXIO):
```

Comment:
Why does it count only these two errors as a failure which can be overcome after retry? Let's invert it to `status not in (0, -errno.ENOENT, -errno.EBADFD)`?